### ISSUE TITLE
Elastic Beanstalk Permissions

### ISSUE THREAD

**Issue by km-pg** [codeship/documentation#338]
I followed the instructions here:
https://codeship.com/documentation/continuous-deployment/deployment-to-elastic-beanstalk/

But, am getting this error:
"A client error (InsufficientPrivilegesException) occurred when calling the UpdateEnvironment operation: You do not have permission to perform the 's3:CreateBucket' action. Verify that your S3 policies and your ACLs allow you to perform these actions."

Maybe the permissions in the documentation need to be updated?

**Comment by km-pg** [comment]
Also just got this error too:
Service:AmazonElasticLoadBalancing, Message:User: <codeship> is not authorized to perform: elasticloadbalancing:RegisterInstancesWithLoadBalancer on resource: arn:aws:elasticloadbalancing:us-east-1:<account id>:loadbalancer/awseb-e-q-AWSEBLoa-...

**Comment by km-pg** [comment]
Also got this error after I added the CreateBucket permission above.

"Service:AmazonCloudFormation, Message:User: <codeship> is not authorized to perform: cloudformation:DescribeStackResources on resource: <cloudformation stack>"

**Comment by mlocher** [comment]
The included permissions are the most basic ones we needed to get a sample application working. Depending on your setup and the services you use, you need to include additional permissions (e.g. CloudFormation).

I'll check our sample app if it is still working an will update the documentation page if required.

**Comment by alemhnan** [comment]
I have the same problem. I followed those instructions [https://codeship.com/documentation/continuous-deployment/deployment-to-elastic-beanstalk/](url) and I get a 
> You do not have permission to perform the 's3:CreateBucket' action.

It does not mention which bucket is trying to create but comparing with another setup (codeship/elasticbeanstalk) that I use with another client seems that the bucket needed is already there. In the setup that is working I have full s3 access for codeship, something that I cannot replicate in this amazon account. 

@mlocher any idea?

**Comment by mlocher** [comment]
@alemhnan the easy solution would be to add the `s3:CreateBucket` permission to the IAM policies. As for where and why your deployment requires this permission, I can't tell. 

Both the bucket needed by the Codeship deployment to store the uploaded archive (this is the one specified in the deployment configuration) as well as another one required by the ElasticBeanstalk application itself is already created? If not, maybe you can temporarily provide this permission and revoke it after the first successful deployment.

Can you run a deployment outside of Codeship with the IAM permissions you have currently configured?

Please note, that (as per our [Contributing Guidelines](https://github.com/codeship/documentation/blob/master/CONTRIBUTING.md#when-not-to-submit-an-issue)) issues on this repository are meant to deal with error in the documentation and should not be used to handle support requests. Please reach out to support@codeship.com in those cases.

**Comment by alemhnan** [comment]
Thanks for the timely answer. I've submitted a PR to the documentation few minutes before you writing. 
To answer your questions: the bucket were already created and usable. I added in the iam role the CreateBucket permission. I am indeed able to deploy from my cli using `eb`
The amazon account was a new one with no prior usage. The only work done was to create the needed buckets and the elasticbeanstalk app so I assumed that the permission in the docs were able to support that minimal use case (they were almost there by the way). 

Thanks for the heads up on the support.
